src/app/pages/TopicPage/Pagination/buildBlocks.js

```diff
@@ -54,22 +52,33 @@ const setDynamicVisibility = state => {
       predicate: e => !e.visibility,
       array: state.result,
     }).forEach(index => {
-      // keeping track of the visibility level we are setting
-      // this will help us determine on which devices we need to display an ellipsis
-      if (index < state.activePageIndex) {
-        state.highestVisibilityOnLeft = deviceSize;
-      } else {
-        state.highestVisibilityOnRight = deviceSize;
-      }
       state.result[index].visibility = deviceSize;
     }),
   );
 
   return state;
+};
 
-  // TODO - if there is just a single number missing at a boundary, fill it in?
-  // eg if we have 1, 2, 3, 5 - should we just add the 4?
-  // Otherwise, we'll have an ellipsis being used to fill in a gap of only one number
+// Ensure we do not have an ellipsis replacing a single number
+const fillEdges = state => {
+  // If page 2 is going to be rendered, it should have the same visibility as page 3
+  const secondElement = state.result[1];
+  const thirdElement = state.result[2];
+  const activePageIsOnRight = state.activePage > 2;
+  if (activePageIsOnRight && secondElement.pageNumber === 2) {
+    secondElement.visibility = thirdElement.visibility;
+  }
+
+  // If page (pagecount - 1) is going to be rendered
+  // it should have the same visibility as page (pagecount - 2)
+  const secondLastElement = state.result[state.result.length - 2];
+  const thirdLastElement = state.result[state.result.length - 3];
+  const secondLastPage = state.pageCount - 1;
+  const activePageIsOnLeft = state.activePage < secondLastPage;
+  if (activePageIsOnLeft && secondLastElement.pageNumber === secondLastPage) {
+    secondLastElement.visibility = thirdLastElement.visibility;
```

Comment:
mutating the state here makes returning the state on line 81 unnecessary. might be better to create a new state object and return that

Reply:
This function is used with `pipe` in the default export (line 165) - it needs to return the reference for use in the next function in the pipe

Reply:
ah ok. I just thought we had an eslint rule to prevent use from making side effects in functions. It's one of the core principles in functional programming. It's also something that was brought up in one of the standards sessions so might be good if we can remove examples other devs might copy from

Reply:
That eslint rule was disabled in this file - re-enabled it and updated all the functions to use clones of state

Reply:
Yas. Pure. Thanks for doing that